store/copr/batch_coprocessor.go

```diff
@@ -827,8 +845,22 @@ func buildBatchCopTasksCore(bo *backoff.Backoffer, store *kvStore, rangesForEach
 			}
 		}
 
+		allTiFlashStores := cache.RegionCache.GetTiFlashStores(tikv.LabelFilterNoTiFlashWriteNode)
+		aliveStores = filterAliveStores(bo.GetCtx(), allTiFlashStores, ttl, store)
+		if careTiFlashReplicaRead {
+			aliveStoreIDsInTiDBZone = make(map[uint64]struct{}, len(aliveStores))
+			for _, as := range aliveStores {
+				// If the `zone` label of the TiFlash store is not set, we treat it as a TiFlash store in other zones.
+				if tiflashZone, isSet := as.GetLabelValue(placement.DCLabelKey); isSet && tiflashZone == tidbZone {
+					aliveStoreIDsInTiDBZone[as.StoreID()] = struct{}{}
```

Comment:
aliveStoreIDsInTiDBZone need reset if needRetry is true.

Reply:
Why? I think it is ok to only init `aliveStoreIDsInTiDBZone` once before the for loop